Tests/BankCardHolderNameTextInputFormatterTests.swift

```diff
@@ -0,0 +1,164 @@
+            editor.insert("-")
+            textInput.expect("A. B-", "", "")
+            editor.insert("C")
+            textInput.expect("A. B-C", "", "")
+        }
+    }
+
+    func testThatDotsAndHyphensAreNotAcceptedAtTheBeginningOfWords() {
+        textInput.edit { editor in
+            editor.insert(".")
+            textInput.expect("", "", "")
+            editor.insert("-")
+            textInput.expect("", "", "")
+            editor.insert("A")
+            textInput.expect("A", "", "")
+            editor.insert(" ")
+            textInput.expect("A ", "", "")
+            editor.insert(".")
+            textInput.expect("A ", "", "")
+            editor.insert("-")
+            textInput.expect("A ", "", "")
+        }
+    }
+
+    func testThatSpacesAreRemovedWhenRequired() {
+        textInput.edit { editor in
+            editor.insert("A B")
+            textInput.expect("A B", "", "")
+        }
+
+        textInput.edit { editor in
+            editor.select(1..<1)
+            editor.backspace()
+            textInput.expect("", "", "B")
+        }
+    }
+
+    func testThatDotsAndSpacesAreRemovedWhenRequired() {
+        textInput.edit { editor in
+            editor.insert("A. B. C.")
+            textInput.expect("A. B. C.", "", "")
+        }
+
+        textInput.edit { editor in
+            editor.select(7..<7)
+            editor.backspace()
+            textInput.expect("A. B. ", "", "")
+
+            editor.select(1..<1)
+            editor.backspace()
+            textInput.expect("", "", "B. ")
+
+            editor.selectAll()
+            editor.insert("A. B. C.")
+            textInput.expect("A. B. C.", "", "")
+
+            editor.select(4..<4)
+            editor.backspace()
+            textInput.expect("A. ", "", "C.")
+        }
+    }
+
+    func testThatHyphensAndSpacesAreRemovedWhenRequired() {
+        textInput.edit { editor in
+            editor.insert("A-B D-E-F")
+            textInput.expect("A-B D-E-F", "", "")
+        }
+
+        textInput.edit { editor in
+            editor.select(1..<1)
+            editor.backspace()
+            textInput.expect("", "", "B D-E-F")
+
+            editor.select(5..<5)
+            editor.backspace()
+            textInput.expect("B D-", "", "F")
+
+            editor.select(3..<3)
+            editor.backspace()
+            textInput.expect("B ", "", "F")
+        }
+    }
+
+    func testThatDigitsAreNotAccepted() {
+        textInput.edit { editor in
+            for digitScalar in UnicodeScalar("0").value...UnicodeScalar("9").value {
+                let digitString = String(UnicodeScalar(digitScalar)!)
+                editor.insert(digitString)
+                textInput.expect("", "", "")
+            }
+        }
+    }
+
+    func testThatDiacriticsIsNotAccepted() {
+        let lettersWithDiacritics = "ÁÀÂÄǍĂĀÃÅǺĆĊĈČĎÉÈĖÊËĚĔĒÍÌİÎÏǏĬĪĨŃN̈ŇÑÓÒÔÖǑŎŌÕŐáàâäǎăāãåǻćċĉčéèėêëěĕēíìîïǐĭīĩńn̈ňñóòôöǒŏōõő"
```

Comment:
This test needs a paired test with only one letter with diacritic (other letters are correct).